### PR TITLE
feat: Implement schema-overlay data models (Issue #6)

### LINKED ISSUE #6: Define data models for medical records
## Requirements
Define Swift data models for medical information. Keep it simple for MVP. 

### Person
- id: UUID
- name: String (encrypted)
- dateOfBirth: Date (encrypted)
- relationship: Enum (self, child, partner, other) - encrypted
- createdAt: Date
- updatedAt: Date

### Attachment
- id: UUID
- recordId: UUID (can link to any record type)
- recordType: Enum (vaccine, condition, medication, etc.) Or can we drop this?
- fileName: String
- fileType: String (e.g., "image/jpeg", "application/pdf")
- encryptedData: Data (the actual encrypted file)
- thumbnailData: Data? (encrypted thumbnail)
- uploadedAt: Date

## User data models

Consider how to make this generic i.e. users can add a record of any arbitrary type, either picking a predefined one (e.g. below), or creating a new one. Perhaps the objects are stored as JSON documents or something? Think about how this works with the "wiki-style history".

### MedicalProvider
- id: UUID
- name: opaque string?
- address? Phone? email? website?
- CreatedAt, UpdatedAt?

### VaccineRecord
- id: UUID
- personId: UUID (FK to Person)
- vaccineName: String (encrypted) - e.g., "MMR", "COVID-19"
- date: Date (encrypted)
- provider: String? (encrypted) - healthcare provider/location? FK to a MedicalProvider?
- Manufacturer?
- batchNumber: String? (encrypted)
- notes: String? (encrypted)
- attachmentIds: [UUID] (references to documents)
- createdAt: Date
- updatedAt: Date

### MedicalCondition
- id: UUID
- personId: UUID
- conditionName: String (encrypted)
- diagnosedDate: Date? (encrypted)
- status: Enum (active, resolved, monitored) - encrypted
- notes: String? (encrypted)
- createdAt: Date
- updatedAt: Date

### Medication
- id: UUID
- personId: UUID
- name: String (encrypted)
- dosage: String? (encrypted)
- frequency: String? (encrypted)
- startDate: Date? (encrypted)
- endDate: Date? (encrypted)
- prescribedBy: String? (encrypted) FK perhaps?
- notes: String? (encrypted)
- createdAt: Date
- updatedAt: Date

## Implementation Notes
- All sensitive fields marked (encrypted) must be encrypted before storage - or the whole record?
- Check whether other fields need to be marked as sensitive
- UUIDs are not sensitive (used for references)
- Timestamps (createdAt, updatedAt) can be plaintext for sorting/filtering
- Consider Codable conformance for serialization
- Relationships: Person 1:N VaccineRecords, 1:N Conditions, 1:N Medications etc.

## Tasks
- [ ] Define model structs/classes
- [ ] Add Codable conformance
- [ ] Document which fields are encrypted
- [ ] Create helper methods for encryption/decryption if needed
- [ ] Add validation methods (e.g., date ranges, required fields)

## Out of Scope
- Sync metadata (Phase 2) but DO consider how to "capture" the arbitrary models/schemas 
- Sharing metadata (Phase 3)
- Version history (future)

## Acceptance Criteria
- All models defined with clear encryption boundaries
- Models conform to Codable
- Code documented with encryption requirements

### PR DESCRIPTION
**Pull request by mnbf9rca**
## Summary

Implements wiki-style schema-overlay architecture for medical records as specified in Issue #6.

**Privacy Enhancement**: SchemaId encrypted inside RecordContent to maintain zero-knowledge server architecture (server cannot infer health information from record types).

### Core Architecture
- **Schema-Overlay Design**: Records store arbitrary `[String: FieldValue]` content with optional schema validation
- **Flexible Person Model**: Uses string array labels instead of fixed relationship enums to avoid nuclear-family assumptions
- **5 Built-in Schemas**: vaccine, condition, medication, allergy, note
- **Custom Schema Support**: Users can create custom schemas or use freeform records
- **Zero-Knowledge Privacy**: SchemaId encrypted with content (server sees only opaque blobs)

### Implementation

**9 Model Files (~1,100 lines)**
- `ModelErrors.swift` - Centralized validation errors
- `FieldValue.swift` - Type-safe dynamic values (string, int, double, bool, date, attachmentIds, stringArray)
- `FieldDefinition.swift` - Schema field specifications with validation rules
- `RecordContent.swift` - Field container with type-safe accessors + encrypted schemaId
- `RecordSchema.swift` - Schema templates
- `BuiltInSchemas.swift` - Predefined schemas (split into extensions)
- `Person.swift` - Individual profiles with flexible labels
- `MedicalRecord.swift` - Record container (opaque encrypted blob only)
- `Attachment.swift` - Attachment metadata with HMAC deduplication

**11 Test Files (~1,900 lines)**
- Comprehensive test coverage for all models
- 457 tests, all passing ✅
- **94.64% overall coverage** (requirement: 90%) ✅
- All files meet 85% threshold ✅

### Documentation
- **ADR-0007**: Schema-Overlay Data Model (58 lines, follows template)
- **ADR-0004**: Updated to document schemaId encryption boundary
- Documents design intent: extensibility, wiki-style approach, flexible person labels, zero-knowledge privacy

## Quality Checks

- ✅ All pre-commit hooks pass (SwiftFormat, SwiftLint, markdownlint)
- ✅ Zero test failures
- ✅ Zero compilation warnings
- ✅ 94.64% test coverage

## Key Design Decisions

1. **Schema-Overlay vs Fixed Types**: Chose flexible schemas as data over fixed content structs for extensibility
2. **Person Labels**: Flexible string arrays (e.g., `["child", "dependent", "legal-guardian"]`) instead of fixed `FamilyRelationship` enum
3. **Built-in Schemas**: Provide sensible defaults while allowing custom user-defined schemas
4. **SchemaId Encryption**: Encrypt schemaId inside RecordContent (not plaintext) to prevent server from inferring health patterns
   - Server cannot distinguish "vaccine" records from "mental-health" records
   - Maintains true zero-knowledge architecture (server sees only opaque encrypted blobs + personId for access control)
5. **Encryption Boundaries Enhanced**: Only id, personId, timestamps, version are plaintext; schemaId + fields are fully encrypted

## Related Issues

Closes #6

## Test Plan

- [x] Unit tests for all models (457 tests)
- [x] 94.64% code coverage achieved
- [x] Pre-commit hooks pass
- [x] Manual verification of schema validation
- [x] Codable round-trip tests for all models
- [x] Encryption boundary tests for MedicalRecord

## Summary by Sourcery

Introduce a schema-overlay data model for medical records with encrypted, schema-aware record content and flexible person labeling, along with comprehensive validation and tests.

New Features:
- Add generic FieldValue, FieldDefinition, RecordContent, RecordSchema, Attachment, Person, and MedicalRecord models to support dynamic, schema-driven medical records with flexible person labels and attachment metadata.
- Provide built-in record schemas for common medical record types including vaccines, conditions, medications, allergies, and general notes.

Privacy Enhancements:
- Encrypt schema identifiers inside RecordContent (not plaintext in MedicalRecord) to prevent server from profiling health data patterns or inferring record types.

Documentation:
- Document the schema-overlay data model design and its trade-offs in a new architecture decision record (ADR-0007).
- Update ADR-0004 to explicitly document schemaId encryption in sync boundaries.

Tests:
- Add extensive unit test coverage for all new models, validation rules, built-in schemas, and Codable/Equatable behavior to ensure correctness of the schema-overlay architecture.

### CODE REVIEW

File: ios/FamilyMedicalApp/FamilyMedicalApp/Models/Attachment.swift [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+        let trimmedFileName = fileName.trimmingCharacters(in: .whitespacesAndNewlines)
+        guard !trimmedFileName.isEmpty else {
+            throw ModelError.fileNameEmpty
+        }
+        guard trimmedFileName.count <= Self.fileNameMaxLength else {
+            throw ModelError.fileNameTooLong(maxLength: Self.fileNameMaxLength)
+        }
+
+        // Validate mimeType
+        guard mimeType.count <= Self.mimeTypeMaxLength else {
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply ios/FamilyMedicalApp/FamilyMedicalApp/Models/Attachment.swift]: Fixed in 7179e61.

Added mimeType trimming to match fileName validation behavior. Now both fields are normalized consistently. Also added test coverage for mimeType whitespace trimming.

File: ios/FamilyMedicalApp/FamilyMedicalApp/Models/Schemas/BuiltInSchemas.swift [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+                    displayOrder: 6,
+                    helpText: "Allergy test results, medical alert bracelet photos, etc."
+                )
+            ],
+            isBuiltIn: true,
+            description: "Allergies and sensitivities"
+        )
+    }
+
+    // MARK: - Note Schema
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply ios/FamilyMedicalApp/FamilyMedicalApp/Models/Schemas/BuiltInSchemas.swift]: Fixed in 95293de.

Removed 4 redundant internal MARK comments. The extension-level MARK comments already provide clear section markers.

File: docs/adr/adr-0007-schema-overlay-data-model.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+- Records store arbitrary `[String: FieldValue]` content
+- Schemas define field structure, validation, and display rules as data
+- Built-in schemas provided for common types (vaccine, medication, condition, allergy, note)
+- Users can create custom schemas or use freeform records (no schema)
+
+Core models:
+
+- `MedicalRecord`: Container with `schemaId` and generic `content`
+- `RecordSchema`: Template defining fields and validation
+- `RecordContent`: Type-safe wrapper for arbitrary fields
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply docs/adr/adr-0007-schema-overlay-data-model.md]: Fixed in 95293de.

Updated ADR-0007 to accurately reflect that MedicalRecord contains encrypted data (opaque blob), and RecordContent (the decrypted payload) contains both schemaId and fields.
